tests/4.5/application_kernels/declare_target_module.F90

```diff
@@ -0,0 +1,52 @@
+!===--- declare_target_module.F90 ------------------------------------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+!
+! This test uses the declare target construct within a subroutine, called
+! in a target region. It checks the result of a simple offloaded addition.
+! This test was provided by the LLNL FGPU repository, derived from the
+! OpenMP 4.5 examples document.
+!
+!===------------------------------------------------------------------------===//
+
+
+#include "ompvv.F90"
+
+#define N 1024
+
+MODULE module_declare
+CONTAINS
+  SUBROUTINE test_declare(x)
+    INTEGER:: x, m
+    !$omp declare target
+    x = x + 1
+  END SUBROUTINE test_declare
+END MODULE module_declare
+
+MODULE params
+  INTEGER:: THRESHOLD = N - 1
+END MODULE params
+
+PROGRAM declare_target_module
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  USE params
+  USE module_declare
+  implicit none
+  INTEGER:: x
+
+  OMPVV_TEST_OFFLOADING
+
+  OMPVV_TEST_SHARED_ENVIRONMENT
+
+  x = N
+
+  !$omp target if(x > THRESHOLD) map(tofrom: x)
```

Comment:
Do we know why the THRESHOLD Thingy?

Reply:
This is from the original 4.5 examples test, https://www.openmp.org/wp-content/uploads/openmp-examples-4.5.0.pdf, pg  115-116 (122-123 on the pdf viewer). It isn't necessary for this test, however.

Reply:
Let's leave it like this and blame it on the Examples